include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -77,8 +79,8 @@ class device_policy
         return ::std::true_type{};
     }
 
-  private:
-    sycl::queue q;
+  protected:
+    mutable ::std::optional<sycl::queue> q;
```

Comment:
I think we might need to include the optional header at the top of this file since I do not think it is included anywhere.

Reply:
Will change.